File: app/src/main/java/hm/orz/chaos114/android/slideviewer/di/InfraModule.java
package hm.orz.chaos114.android.slideviewer.di;

import android.app.Application;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import hm.orz.chaos114.android.slideviewer.infra.network.SlideShareLoader;
import hm.orz.chaos114.android.slideviewer.infra.repository.TalkRepository;

@Module
public class InfraModule {
    @Singleton
    @Provides
    public TalkRepository provideTalkRepository(Application app) {
        return new TalkRepository(app);
    }

    @Singleton
    @Provides
    public SlideShareLoader provideSlideShareLoader(TalkRepository talkRepository) {
        return new SlideShareLoader(talkRepository);
    }
}

File: app/src/main/java/hm/orz/chaos114/android/slideviewer/di/InfraModule.kt
package hm.orz.chaos114.android.slideviewer.di

import android.app.Application
import dagger.Module
import dagger.Provides
import hm.orz.chaos114.android.slideviewer.infra.network.SlideShareLoader
import hm.orz.chaos114.android.slideviewer.infra.repository.SettingsRepository
import hm.orz.chaos114.android.slideviewer.infra.repository.TalkRepository
import javax.inject.Singleton

@Module
class InfraModule {
    @Singleton
    @Provides
    fun provideTalkRepository(app: Application): TalkRepository {
        return TalkRepository(app)
    }

    @Singleton
    @Provides
    fun provideSettingsRepository(app: Application): SettingsRepository {
        return SettingsRepository(app)
    }

    @Singleton
    @Provides
    fun provideSlideShareLoader(talkRepository: TalkRepository): SlideShareLoader {
        return SlideShareLoader(talkRepository)
    }
}
